### PR TITLE
swagger_controller DSL can accept a resource_path which will be used over a generated path

### PR DESCRIPTION
**Pull request by sb8244**
This allows for custom paths to be defined which can then be fed into swagger-ui in a more user friendly way.

### CODE REVIEW

File: spec/fixtures/controllers/custom_resource_path_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module Api
+  module V1
+    class SuperclassController < ApplicationController
+    end
+    class CustomResourcePathController < SuperclassController
+      swagger_controller :custom_resource_path, "User Management", resource_path: "testing"
```
Comment (sb8244) [inline spec/fixtures/controllers/custom_resource_path_controller.rb]: Here is the proposed change

**Comment by sb8244** [comment]
This may not work like I expected. I put this into my project and swagger 1.2 -> 2.0 replaces `nested/testing/resource` as `resource`, returning everything after the final /.

However, swagger 1.2 generates a document with the correct `nested/testing/resource`, which makes me think the problem is in swagger2 converter and not this change.

**Comment by richhollis** [comment]
Thanks for this! I've now manually merged this into the latest version 0.2.6. Cheers.

**Comment by sb8244** [comment]
Awesome thank you! Swagger docs has allowed us to have a successful automated doc system! Great stuff

**Comment by richhollis** [comment]
Thank you for the PR! Great to hear it's worked out for you guys.

**Pull request closed without merging**